toolkit/tools/imagecustomizerapi/overlay.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Overlay struct {
+	LowerDir  string           `yaml:"LowerDir"`
+	UpperDir  string           `yaml:"UpperDir"`
+	WorkDir   string           `yaml:"WorkDir"`
+	Partition *VerityPartition `yaml:"Partition"`
+}
+
+func (o *Overlay) IsValid() error {
+	// Validate paths for UpperDir, WorkDir, and LowerDir
+	if err := validatePath(o.UpperDir); err != nil {
+		return fmt.Errorf("UpperDir: %v", err)
+	}
+	if err := validatePath(o.WorkDir); err != nil {
+		return fmt.Errorf("WorkDir: %v", err)
+	}
+	if err := validatePath(o.LowerDir); err != nil {
+		return fmt.Errorf("LowerDir: %v", err)
+	}
+
+	// Check if UpperDir and WorkDir are identical
+	if o.UpperDir == o.WorkDir {
+		return fmt.Errorf("UpperDir and WorkDir must be distinct")
```

Comment:
should we include the values of these directories in the error messages here? If i have multiple overlays, and i get this error, it'd be nice if it points me directly to which one has the problem...

Reply:
Fixed.